### PR TITLE
f-header@2.0.0-beta.1 logo component + tenants resource files

### PR DESCRIPTION
**Pull request by dandel10n**
### Added
- Logo component
- Tenants resource files
- Vue.config.js for rewriting default vue cli webpack config
- common.scss base styles imports
- theme and locale logic for Header component (later will be moved to independant package because it repeats in footer and searchbox already)

### CODE REVIEW

File: packages/f-header/src/components/Header.vue [hunk 1]
```diff
@@ -1,15 +1,70 @@
...
+            default: ''
+        },
+        isTransparent: {
+            type: Boolean,
+            required: true
+        }
+    },
+    data () {
+        const locale = this.getLocale();
+        const localeConfig = tenantConfigs[locale];
```
Comment (markedup) [inline packages/f-header/src/components/Header.vue:34]: Could this be set with the fallback inline here?
```
const localeConfig = tenantConfigs[locale] || 'en-GB';
```
…and therefore remove the necessity for the check on line 50?

**Review by BanksySan — APPROVED** [review]

File: packages/f-header/src/components/Logo.vue [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            return `${this.theme}-icon`;
+        },
+        linkAltText () {
+            return `Go to ${this.companyName} homepage`;
+        }
+    }
+};
+</script>
+
+<style lang="scss" scoped>
```
Comment (zlatinivanov-je) [inline packages/f-header/src/components/Logo.vue]: Does it make sense to have a separate SCSS file here?

File: packages/f-header/src/tenants/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import uk from './en-GB';
```
Comment (zlatinivanov-je) [inline packages/f-header/src/tenants/index.js:1]: Is this file going to include all tenant translations in the component?

**Review by zlatinivanov-je — APPROVED** [review]
A couple of minor questions. But really happy to see work done on this... will try on my project tomorrow. 👍

File: packages/f-header/src/components/Logo.vue (hunk 2 above)
Comment (dandel10n) [reply packages/f-header/src/components/Logo.vue]: It is more visible to have component's styles in the same file as component's logic.  Actually I had a plan to change it to module styles according to this [discussion](https://github.com/justeat/fozzie-components/issues/42#issuecomment-528323259)

File: packages/f-header/src/tenants/index.js (hunk 3 above)
Comment (dandel10n) [reply packages/f-header/src/tenants/index.js:1]: Yes, but they are only around 400 bytes each. I am open to any suggestions on how we can optimize it

**Comment by dandel10n** [comment]
> A couple of minor questions. But really happy to see work done on this... will try on my project tomorrow. 👍

@zlatinivanov-je thanks! to early to try, because there is only logo in the header component at the moment, just started working on it yesterday. But hope to publish it very soon!

**Review by markedup — COMMENTED** [review]
Minor query, otherwise looks good!

File: packages/f-header/src/components/Header.vue (hunk 1 above)
Comment (dandel10n) [reply packages/f-header/src/components/Header.vue:34]: `tenantConfigs[locale]` returns whole resource object for specific tenant, which then written to `copy` vue data, not just locale.

File: packages/f-header/src/components/Header.vue [hunk 4]
```diff
@@ -1,15 +1,70 @@
 <template>
-    <header class="c-header">
-        I am Header
+    <header
+        :data-theme="theme"
+        class="c-header">
+        <logo
+            :theme="theme"
+            :is-transparent=isTransparent
```
Comment (DamianMullins) [inline packages/f-header/src/components/Header.vue]: `isTransparent` should be in quotes.

File: packages/f-header/src/components/Demo.vue [hunk 5]
```diff
@@ -7,6 +7,7 @@
             name="viewport"
             content="width=device-width, initial-scale=1">
         <vue-header
+            :is-transparent="true"
```
Comment (DamianMullins) [inline packages/f-header/src/components/Demo.vue:10]: Can just do `is-transparent` here.

Comment (dandel10n) [reply packages/f-header/src/components/Demo.vue:10]: Yes, but it is only demo file where you can change it to false for testing purposes. So just faster to change true to false

File: packages/f-header/src/components/Logo.vue [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                }
+            }
+        }
+    }
+
+    .c-icon--je g {
+        fill: $header-logo-color;
+    }
+
+    .c-icon-je--transparentBckgrnd g {
```
Comment (DamianMullins) [inline packages/f-header/src/components/Logo.vue]: I think `c-icon-je--transparentBg` might be a better class name?

File: packages/f-header/src/components/tests/Header.test.js [hunk 7]
```diff
@@ -3,15 +3,58 @@ import Header from '../Header.vue';
...
+        };
+        const wrapper = shallowMount(Header, { propsData });
 
         // Assert
         expect(wrapper).toMatchSnapshot();
     });
+
+
+    it('should render ml themed component if AU local passed', () => {
+        // Arrange & Act
```
Comment (DamianMullins) [inline packages/f-header/src/components/tests/Header.test.js]: Not a big deal but these joined "Arrange & Act" sections can be split out into separate "Arrange" and "Act" sections.

File: packages/f-header/src/components/Logo.vue (hunk 6 above)
Comment (dandel10n) [reply packages/f-header/src/components/Logo.vue]: ✅

File: packages/f-header/src/components/tests/Logo.test.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import { shallowMount } from '@vue/test-utils';
+import Logo from '../Logo.vue';
+// import {
```
Comment (DamianMullins) [inline packages/f-header/src/components/tests/Logo.test.js]: Can this comment block be removed?

Comment (dandel10n) [reply packages/f-header/src/components/tests/Logo.test.js]: ✅

**Review by DamianMullins — APPROVED** [review]

File: packages/f-header/src/assets/scss/common.scss [hunk 9]
```diff
@@ -0,0 +1,50 @@
+// Set $theme variable so that fozzie doesn't complain when vars are pulled in
+// n.b. This isn't used currently (unless there are theme specific vars from fozzie)
+// The theme is handled via a data-attribute on the component
+$theme: 'je' !default;
+
+@import '~@justeat/f-utils/src/scss/functions/index';
```
Comment (ashleynolan) [inline packages/f-header/src/assets/scss/common.scss:6]: Late comment (just a small optimization) – you should just be able to import `f-utils` base `index.scss` here, as it includes the functions and mixins. Will save you 1 whole import line 😉

Comment (dandel10n) [reply packages/f-header/src/assets/scss/common.scss:6]: Will add this change in my next PR :+1:

Comment (dandel10n) [reply packages/f-header/src/assets/scss/common.scss:6]: It looks like `@import`s inside `@justeat/f-utils/src/scss/index` are not recognised on the projects level. Receive an error:

```
@import 'mixins';
^
File to import not found or unreadable: mixins.
```

Comment (ashleynolan) [reply packages/f-header/src/assets/scss/common.scss:6]: I think I know why this is – I think it expects it to be `mixins/` annoyingly or something like that (if I remember right).

Could change this locally in your `node_modules` and see if that's right – if it is, can change it in `f-utils`…

**Pull request merged**